### ISSUE TITLE
rabbitmq_policy action set is not seen as idempotent by Chef w.r.t. resource execution

### ISSUE THREAD

**Issue by gordonbondon** [rabbitmq/chef-cookbook#500]
Cookbook version: 5.6.1

#425 removed check for policy existance before set_policy is used. This has fixed the issue when policy was not reapplied on parameter changes. But now it is causing similar messages to appear every chef run:
```
* rabbitmq_policy[ha-all] action set
           * execute[set_policy ha-all] action run
             - execute rabbitmqctl set_policy -p / ha-all ".*" '{"ha-mode":"exactly","ha-params":2,"ha-sync-mode":"automatic","queue-master-locator":"min-masters"}' --priority 0
```

Which in turn fails if you are using `enforce_idempotency` in your test kitchen suite.

Will you accept a pull request with a change: compare configs and use `set_policy` only when it has changed?

**Comment by michaelklishin** [comment]
You can try comparing defined policies. The rest of configuration is orthogonal to policies. I have concerns that this may end up being too fragile but let's wait and see.

**Comment by michaelklishin** [comment]
This should be easier if there was a way to display definition of a single policy e.g. as JSON. I will look at extending `rabbitmq-diagnostics` to that.

**Comment by fr33l** [comment]
any update on this please?

**Comment by michaelklishin** [comment]
@fr33l there were no updates. The proposed contribution has never materialized. This is open source software, so you are welcome to look into addressing this. Asking others to do it for you is not how it works around here.

**Comment by michaelklishin** [comment]
This should be addressed by #587.
